Index: Components/RTShaderSystem/CMakeLists.txt
===================================================================
--- Components/RTShaderSystem/CMakeLists.txt	(revision 669)
+++ Components/RTShaderSystem/CMakeLists.txt	(working copy)
@@ -77,15 +77,10 @@
   src/OgreShaderScriptTranslator.cpp
   src/OgreShaderMaterialSerializerListener.cpp
   src/OgreShaderProgramProcessor.cpp
-  src/OgreShaderCGProgramProcessor.cpp
-  src/OgreShaderCGProgramWriter.cpp
-  src/OgreShaderGLSLProgramProcessor.cpp
-  src/OgreShaderGLSLProgramWriter.cpp
+  
   src/OgreShaderGLSLESProgramProcessor.cpp
   src/OgreShaderGLSLESProgramWriter.cpp
   src/OgreShaderProgramWriterManager.cpp
-  src/OgreShaderHLSLProgramProcessor.cpp
-  src/OgreShaderHLSLProgramWriter.cpp
   src/OgreShaderExLayeredBlending.cpp
   src/OgreShaderExHardwareSkinning.cpp
   src/OgreShaderExHardwareSkinningTechnique.cpp
@@ -94,6 +89,16 @@
   src/OgreShaderExTextureAtlasSampler.cpp
 )
 
+if (NOT ORKIGE_BUILD_ANDROID)
+  list(APPEND SOURCE_FILES
+    src/OgreShaderGLSLProgramProcessor.cpp
+    src/OgreShaderGLSLProgramWriter.cpp  
+    src/OgreShaderCGProgramProcessor.cpp
+    src/OgreShaderCGProgramWriter.cpp
+    src/OgreShaderHLSLProgramProcessor.cpp
+    src/OgreShaderHLSLProgramWriter.cpp
+  )
+endif()  
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
Index: Components/RTShaderSystem/include/OgreShaderGLSLESProgramWriter.h
===================================================================
--- Components/RTShaderSystem/include/OgreShaderGLSLESProgramWriter.h	(revision 669)
+++ Components/RTShaderSystem/include/OgreShaderGLSLESProgramWriter.h	(working copy)
@@ -147,7 +147,8 @@
     StringMap                   mDefinesMap;                    // Map of #defines and the function library that contains them
 	ParamContentToStringMap		mContentToPerVertexAttributes;	// Map parameter content to vertex attributes
 	int							mGLSLVersion;					// Holds the current glsl es version
-	StringVector				mFragInputParams;				// Holds the fragment input params 
+	StringVector				mVaryingParams;					// Holds the varying params 
+	StringVector				mOutVaryingParams;				// Holds the varying params whic are used as out paramater
     StringMap                   mCachedFunctionLibraries;       // Holds the cached function libraries
 };
 
Index: Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp
===================================================================
--- Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp	(revision 669)
+++ Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp	(working copy)
@@ -259,9 +259,10 @@
                     "GLSLESProgramWriter::writeSourceCode" );	
             }
 
-            // Clear out old input params
-            mFragInputParams.clear();
-            
+            // Clear out old varying params
+            mVaryingParams.clear();
+            mOutVaryingParams.clear();
+			
             const ShaderFunctionList& functionList = program->getFunctions();
             ShaderFunctionConstIterator itFunction;
 
@@ -374,37 +375,35 @@
                         Parameter::Content content = op.getParameter()->getContent();
 
                         // Check if we write to a varying because the are only readable in fragment programs 
-                        if (opSemantic == Operand::OPS_OUT || opSemantic == Operand::OPS_INOUT)
+                        if (opSemantic == Operand::OPS_OUT)
                         {	
                             // Is the written parameter a varying 
                             bool isVarying = false;
 
-                            // Check if we write to a varying because the are only readable in fragment programs 
-                            if (gpuType == GPT_FRAGMENT_PROGRAM)
-                            {	
-                                StringVector::iterator itFound = std::find(mFragInputParams.begin(), mFragInputParams.end(), paramName);	
-                                if(itFound != mFragInputParams.end())
-                                {						
-                                    // Declare the copy variable
-                                    String newVar = "local_" + paramName;
-                                    String tempVar = paramName;
-                                    isVarying = true;
+							StringVector::iterator itFound = std::find(mVaryingParams.begin(), mVaryingParams.end(), paramName);	
+							if(itFound != mVaryingParams.end())
+							{						
+								// Declare the copy variable
+								String newVar = "local_" + paramName;
+								String tempVar = paramName;
+								isVarying = true;
 
-                                    // We stored the original values in the mFragInputParams thats why we have to replace the first var with o
-                                    // because all vertex output vars are prefixed with o in glsl the name has to match in the fragment program.
-                                    tempVar.replace(tempVar.begin(), tempVar.begin() + 1, "o");
+								// We stored the original values in the mVaryingParams thats why we have to replace the first var with o
+								// because all vertex output vars are prefixed with o in glsl the name has to match in the fragment program.
+								tempVar.replace(tempVar.begin(), tempVar.begin() + 1, "o");
 
-                                    // Declare the copy variable and assign the original
-                                    os << "\t" << mGpuConstTypeMap[op.getParameter()->getType()] << " " << newVar << " = " << tempVar << ";\n" << ENDL;	
+								// Declare the copy variable and assign the original
+								os << "\t" << mGpuConstTypeMap[op.getParameter()->getType()] << " " << newVar << " = " << tempVar << ";\n" << ENDL;	
 
-                                    // From now on we replace it automatic 
-                                    mInputToGLStatesMap[paramName] = newVar;
+								// From now on we replace it automatic 
+								mInputToGLStatesMap[paramName] = newVar;
 
-                                    // Remove the param because now it is replaced automatic with the local variable
-                                    // (which could be written).
-                                    mFragInputParams.erase(itFound++);
-                                }
-                            }
+								// Remove the param because now it is replaced automatic with the local variable
+								// (which could be written).
+								mVaryingParams.erase(itFound++);
+								// Hold the param to remember to set its value in the end of the shader
+								mOutVaryingParams.push_back(paramName);
+							}
                             
                             // If its not a varying param check if a uniform is written
                             if(!isVarying)
@@ -538,9 +537,11 @@
                     os << "\tgl_FragColor = outputColor;" << ENDL;
                 }
                 else if (gpuType == GPT_VERTEX_PROGRAM)
-                {
-                    os << "\tgl_Position = outputPosition;" << ENDL;
-                }
+				{
+					os << "\tgl_Position=outputPosition;" << ENDL;
+					for (StringVector::iterator it = mOutVaryingParams.begin(); it != mOutVaryingParams.end(); it++)
+							os << "\t" << *it << "=" << mInputToGLStatesMap[*it] << ";" << ENDL;
+				}
 
                 os << "}" << ENDL;
             }
@@ -572,7 +573,7 @@
                 {					
                     // push fragment inputs they all could be written (in glsl you can not write
                     // input params in the fragment program)
-                    mFragInputParams.push_back(paramName);
+                    mVaryingParams.push_back(paramName);
 
                     // In the vertex and fragment program the variable names must match.
                     // Unfortunately now the input params are prefixed with an 'i' and output params with 'o'.
@@ -662,6 +663,7 @@
 							os << "[" << pParam->getSize() << "]";	
 						}
 						os << ";" << ENDL;	
+						mVaryingParams.push_back(pParam->getName());
                     }
                 }
                 else if(gpuType == GPT_FRAGMENT_PROGRAM &&
Index: OgreMain/CMakeLists.txt
===================================================================
--- OgreMain/CMakeLists.txt	(revision 669)
+++ OgreMain/CMakeLists.txt	(working copy)
@@ -12,6 +12,10 @@
 ############################################################
 
 PROJECT(OgreMain)
+include_directories(${ORKIGE_FREEIMAGE_INCLUDE})
+include_directories(${ORKIGE_FREETYPE_INCLUDE})
+include_directories(${ORKIGE_ZLIB_INCLUDE})
+include_directories(${ORKIGE_ZZIP_INCLUDE})
 
 include(PrecompiledHeader)
 
@@ -40,7 +44,7 @@
   include/OgreBlendMode.h
   include/OgreBone.h
   include/OgreBorderPanelOverlayElement.h
-  ${OGRE_BINARY_DIR}/include/OgreBuildSettings.h
+  ${OGRE_SOURCE_DIR}/Settings/OgreBuildSettings.h
   include/OgreCamera.h
   include/OgreCodec.h
   include/OgreColourValue.h
@@ -574,6 +578,29 @@
     set(PLATFORM_LIBS ${Carbon_LIBRARY_FWK} ${Cocoa_LIBRARY_FWK})
     set(PLATFORM_HEADER_INSTALL "OSX")
   endif ()
+elseif(ORKIGE_BUILD_ANDROID)
+    include_directories(include/Android)
+    set(PLATFORM_HEADER_FILES
+      include/Android/OgreConfigDialogImp.h
+      include/Android/OgreErrorDialogImp.h
+      include/Android/OgreTimerImp.h
+      include/OgreStringSerialiser.h
+    )
+    set(PLATFORM_SOURCE_FILES
+      src/Android/OgreConfigDialog.cpp
+      src/Android/OgreErrorDialog.cpp
+      src/Android/OgreTimer.cpp
+      src/OgreSearchOps.cpp
+      src/OgreStringSerialiser.cpp
+    )
+    set(PLATFORM_LIBS dl)
+    set(PLATFORM_HEADER_INSTALL "Android")
+    # some platforms require explicit linking to libdl, see if it's available
+    find_library(DL_LIBRARY NAMES dl dlsym)
+    mark_as_advanced(DL_LIBRARY)
+    if (DL_LIBRARY)
+      list(APPEND PLATFORM_LIBS dl)
+    endif ()
 elseif (UNIX)
   include_directories(include/GLX ${X11_INCLUDE_DIR})
   set(PLATFORM_HEADER_FILES
@@ -648,7 +675,7 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src/nedmalloc)
 
 set(LIBRARIES
-  ${FREETYPE_LIBRARIES}
+  ${ORKIGE_FREETYPE_TARGET}
   ${PLATFORM_LIBS}
 )
 
@@ -659,7 +686,7 @@
 if (OGRE_CONFIG_ENABLE_FREEIMAGE)
   list(APPEND HEADER_FILES include/OgreFreeImageCodec.h)
   list(APPEND SOURCE_FILES src/OgreFreeImageCodec.cpp)
-  list(APPEND LIBRARIES    "${FreeImage_LIBRARIES}")
+  list(APPEND LIBRARIES    "${ORKIGE_FREEIMAGE_TARGET}")
 endif()
 
 if (OGRE_CONFIG_ENABLE_DDS)
@@ -675,8 +702,8 @@
 if (OGRE_CONFIG_ENABLE_ZIP)
   list(APPEND HEADER_FILES include/OgreZip.h)
   list(APPEND SOURCE_FILES src/OgreZip.cpp)
-  list(APPEND LIBRARIES "${ZZip_LIBRARIES}")
-  list(APPEND LIBRARIES "${ZLIB_LIBRARIES}")
+  list(APPEND LIBRARIES "${ORKIGE_ZZIP_TARGET}")
+  list(APPEND LIBRARIES "${ORKIGE_ZLIB_TARGET}")
 endif ()
 
 if (OGRE_CONFIG_ENABLE_GLES2_GLSL_OPTIMISER)
Index: RenderSystems/GLES2/CMakeLists.txt
===================================================================
--- RenderSystems/GLES2/CMakeLists.txt	(revision 669)
+++ RenderSystems/GLES2/CMakeLists.txt	(working copy)
@@ -93,7 +93,22 @@
 endif (OGRE_CONFIG_ENABLE_GLES2_CG_SUPPORT)
 
 # Add system specific settings
-if (OGRE_BUILD_PLATFORM_NACL)
+if (ORKIGE_BUILD_ANDROID)
+#message("ANDROID" ${OPENGLES2_gl_LIBRARY})
+  set(PLATFORM_HEADERS
+    include/Android/OgreAndroidGLContext.h
+    include/Android/OgreAndroidGLSupport.h
+    include/Android/OgreAndroidWindow.h
+    include/Android/OgreGLES2Util.h
+  )
+  set(PLATFORM_SOURCES
+    src/Android/OgreAndroidGLContext.cpp
+    src/Android/OgreAndroidGLSupport.cpp
+    src/Android/OgreAndroidWindow.cpp
+  )
+  include_directories(src/Android include/Android)
+  set(PLATFORM_HEADER_INSTALL "Android")
+elseif (OGRE_BUILD_PLATFORM_NACL)
   set(PLATFORM_HEADERS 
     include/NaCl/OgreGLES2Util.h
     include/NaCl/OgreNaClGLContext.h
Index: RenderSystems/GLES2/include/OgreGLES2Prerequisites.h
===================================================================
--- RenderSystems/GLES2/include/OgreGLES2Prerequisites.h	(revision 669)
+++ RenderSystems/GLES2/include/OgreGLES2Prerequisites.h	(working copy)
@@ -48,7 +48,7 @@
 #ifndef GL_GLEXT_PROTOTYPES
 #  define  GL_GLEXT_PROTOTYPES
 #endif
-
+#		define GL_OES_vertex_array_object 0
 #if (OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS)
 #   include <OpenGLES/ES2/gl.h>
 #   include <OpenGLES/ES2/glext.h>
@@ -56,6 +56,9 @@
 #       include <OpenGLES/EAGL.h>
 #   endif
 #elif (OGRE_PLATFORM == OGRE_PLATFORM_ANDROID) || (OGRE_PLATFORM == OGRE_PLATFORM_NACL)
+#	if __ANDROID_API__ <= 8
+#		define GL_OES_vertex_array_object 0
+#	endif
 #	ifndef GL_GLEXT_PROTOTYPES
 #		define  GL_GLEXT_PROTOTYPES
 #	endif
Index: RenderSystems/GLES2/src/Android/OgreAndroidGLSupport.cpp
===================================================================
--- RenderSystems/GLES2/src/Android/OgreAndroidGLSupport.cpp	(revision 669)
+++ RenderSystems/GLES2/src/Android/OgreAndroidGLSupport.cpp	(working copy)
@@ -38,8 +38,13 @@
 #include "OgreAndroidWindow.h"
 #include "OgreAndroidGLContext.h"
 
+
+
 namespace Ogre {
 
+	int GetAndroidWindowWidth();
+		int GetAndroidWindowHeight();
+
     AndroidGLSupport::AndroidGLSupport()
     {
     }
@@ -75,7 +80,8 @@
             NameValuePairList miscParams;
 
             bool fullscreen = true;
-			unsigned int w = 800, h = 600;
+			unsigned int w = GetAndroidWindowWidth();
+			unsigned int h = GetAndroidWindowHeight();
 
             if ((opt = mOptions.find("Display Frequency")) != end)
             {
Index: RenderSystems/GLES2/src/Android/OgreAndroidWindow.cpp
===================================================================
--- RenderSystems/GLES2/src/Android/OgreAndroidWindow.cpp	(revision 669)
+++ RenderSystems/GLES2/src/Android/OgreAndroidWindow.cpp	(working copy)
@@ -101,6 +101,9 @@
 				ctxHandle = Ogre::StringConverter::parseInt(opt->second);
 			}
 			
+			mHandle = 0;
+			ctxHandle = 0;
+			
 			if(ctxHandle != -1)
 			{
 				mContext = new AndroidGLContext(mGLSupport, ctxHandle);
@@ -117,16 +120,24 @@
 	void AndroidWindow::createNativeWindow( int &left, int &top, uint &width, uint &height, String &title )
 	{
 		LogManager::getSingleton().logMessage("\tcreateNativeWindow called");
+        mLeft = left;
+        mTop = top;
+        mWidth = width;
+        mHeight = height;
 	}
 
 	void AndroidWindow::reposition( int left, int top )
 	{
 		LogManager::getSingleton().logMessage("\treposition called");
+        mLeft = left;
+        mTop = top;
 	}
 
 	void AndroidWindow::resize(uint width, uint height)
 	{
 		LogManager::getSingleton().logMessage("\tresize called");
+        mWidth = width;
+        mHeight = height;
 	}
 
 	void AndroidWindow::windowMovedOrResized()
Index: RenderSystems/GLES2/src/OgreGLES2HardwarePixelBuffer.cpp
===================================================================
--- RenderSystems/GLES2/src/OgreGLES2HardwarePixelBuffer.cpp	(revision 669)
+++ RenderSystems/GLES2/src/OgreGLES2HardwarePixelBuffer.cpp	(working copy)
@@ -154,11 +154,13 @@
             allocateBuffer();
             // No scaling or conversion needed
             scaled = src;
+#if OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS
             if (src.format == PF_R8G8B8)
             {
                 scaled.format = PF_B8G8R8;
                 PixelUtil::bulkPixelConversion(src, scaled);
             }
+#endif
         }
 
         upload(scaled, dstBox);
Index: RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp
===================================================================
--- RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp	(revision 669)
+++ RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp	(working copy)
@@ -89,7 +89,11 @@
             case PF_A8B8G8R8:
             case PF_B8G8R8A8:
             case PF_A2R10G10B10:
+#if OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS
                 return GL_BGRA;
+#else
+                return GL_RGBA;
+#endif
 			case PF_X8B8G8R8:
 			case PF_R8G8B8A8:
             case PF_A2B10G10R10:
Index: RenderSystems/GLES2/src/OgreGLES2RenderSystem.cpp
===================================================================
--- RenderSystems/GLES2/src/OgreGLES2RenderSystem.cpp	(revision 669)
+++ RenderSystems/GLES2/src/OgreGLES2RenderSystem.cpp	(working copy)
@@ -164,12 +164,13 @@
     {
 		mGLSupport->start();
 
-        // Create the texture manager        
-		mTextureManager = OGRE_NEW GLES2TextureManager(*mGLSupport); 
-
         RenderWindow *autoWindow = mGLSupport->createWindow(autoCreateWindow,
                                                             this, windowTitle);
         RenderSystem::_initialise(autoCreateWindow, windowTitle);
+
+		// Create the texture manager        
+		mTextureManager = OGRE_NEW GLES2TextureManager(*mGLSupport); 
+
         return autoWindow;
     }
 
Index: RenderSystems/GLES/CMakeLists.txt
===================================================================
--- RenderSystems/GLES/CMakeLists.txt	(revision 669)
+++ RenderSystems/GLES/CMakeLists.txt	(working copy)
@@ -105,6 +105,21 @@
   include_directories(src/EAGL include/EAGL)
   link_directories(${OPENGLES_LIBRARY_PATH})
   set(PLATFORM_HEADER_INSTALL "APPLE")
+elseif (ORKIGE_BUILD_ANDROID)
+  set(PLATFORM_HEADERS
+    include/Android/OgreAndroidGLESContext.h
+    include/Android/OgreAndroidGLESSupport.h
+    include/Android/OgreAndroidGLESWindow.h
+    include/Android/OgreAndroidGLESUtil.h
+  )
+  set(PLATFORM_SOURCES
+    src/Android/OgreAndroidGLESContext.cpp
+    src/Android/OgreAndroidGLESSupport.cpp
+    src/Android/OgreAndroidGLESWindow.cpp
+  )
+  include_directories(src/Android include/Android)
+  set(PLATFORM_HEADER_INSTALL "Android")
+  set(PLATFORM_LIBS )
 elseif (UNIX)
   set(PLATFORM_HEADERS
     include/EGL/X11/OgreX11EGLContext.h
